console.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/miquella/vaulted/lib"
+	"github.com/pkg/browser"
+)
+
+var (
+	ErrInvalidTemporaryCredentials = errors.New("Temporary session credentials found. Console cannot be opened with temporary credentials.\nIf the AWS Key is set in a vault, the permanent credentials can be used by specifying the vault name.")
+	ErrInvalidDuration             = errors.New("Console duration must be between 15m and 12h.")
+	ErrNoCredentialsFound          = errors.New("No credentials found. Console cannot be opened.")
+)
+
+const (
+	ConsoleURL                 = "https://console.aws.amazon.com/console/home"
+	ConsoleFederationSigninURL = "https://signin.aws.amazon.com/federation"
+
+	ConsoleMinDuration     = 15 * time.Minute
+	ConsoleMaxDuration     = 12 * time.Hour
+	ConsoleDefaultDuration = 1 * time.Hour
+)
+
+type Console struct {
+	VaultName string
+	Role      string
+	Duration  time.Duration
+}
+
+func (c *Console) Run(store vaulted.Store) error {
+	signinToken, err := c.getSigninToken(store)
+	if err != nil {
+		return err
+	}
+
+	return openConsole(signinToken)
+}
+
+func (c *Console) getSigninToken(store vaulted.Store) (string, error) {
+	vault, err := c.getVault(store)
```

Comment:
It feels like we really should not be storing and passing on a vault to everything in this function.

It feels like we should be calling OpenVault when necessary and pulling the requisite pieces (Key, Duration, etc) long enough to store _those_ but not holding and sharing the vault with everything else to get their piece from it.

It feels like this level should have some sort of control flow on `if c.VaultName != ""`, rather than doing this down in another method.